pinot-query-runtime/src/test/resources/queries/TypeCasting.json

```diff
@@ -0,0 +1,163 @@
+{
+  "explicit_cast": {
+    "comment": "explicit type casting for operations",
```

Comment:
I thought `explicit` is specifically when user has requested via CAST function. `Implicit` should also be tested right ?

Reply:
yes. the below 2 test categories are implicit casting. they are split into 2:
- implicit cast b/c planner derives the cast type (based on SqlStdOperatorTable)
- implicit cast b/c Pinot's function catalog requires implicit casting (based on FunctionRegistry)